# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/nsNSSIOLayer.cpp
# Commit: c922a30d444e
# Full Hash: c922a30d444e359e6fa572089eb29b85e09a1168
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2021-09-17 21:50:08
# Regressor Bug: 1382886
# File Overlap Count: 1
# Description:
#   Bug 1382886 - Make sure that nsHttpConnection immediately knows that the handshake is done. r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123919
# ==============================================================================

diff -r 7f2260b49e60 -r c922a30d444e security/manager/ssl/nsNSSIOLayer.cpp
--- a/security/manager/ssl/nsNSSIOLayer.cpp	Fri Sep 17 13:19:01 2021 +0000
+++ b/security/manager/ssl/nsNSSIOLayer.cpp	Fri Sep 17 13:19:01 2021 +0000
@@ -251,17 +251,10 @@
 
   mIsFullHandshake = false;  // reset for next handshake on this connection
 
-  // HandshakeDone needs to be dispatched so that it is not called inside
-  // nss locks.
-  RefPtr<nsNSSSocketInfo> self(this);
-  NS_DispatchToCurrentThread(NS_NewRunnableFunction(
-      "nsNSSSocketInfo::HandshakeDone",
-      [self{std::move(self)}]() {
-        if (self->mTlsHandshakeCallback) {
-          auto callback = std::move(self->mTlsHandshakeCallback);
-          Unused << callback->HandshakeDone();
-        }
-      }));
+  if (mTlsHandshakeCallback) {
+    auto callback = std::move(mTlsHandshakeCallback);
+    Unused << callback->HandshakeDone();
+  }
 }
 
 void nsNSSSocketInfo::SetNegotiatedNPN(const char* value, uint32_t length) {
